scapy/contrib/isotp.py

```diff
@@ -1611,3 +1613,368 @@ def recv(self, x=0xffff):
+        packet: packet to send
+        number_of_packets: number of packets send
+
+    This function is used for scanning with extended addresses.
+    It sends multiple packets at once. The number of packets
+    is defined in the number_of_packets variable.
+    It only iterates the extended ID, NOT the actual ID of the packet.
+    This method is used in extended scan function.
+    """
+    end_id = min(ext_id + number_of_packets, 255)
+    for i in range(ext_id, end_id + 1):
+        packet.extended_address = i
+        sock.send(packet)
+
+
+def get_isotp_packet(identifier=0x0, extended=False):
+    """ Craft ISO TP packet
+    Args:
+        identifier: identifier of crafted packet
+        extended: boolean if packet uses extended address
+    """
+
+    if extended:
+        pkt = ISOTPHeaderEA() / ISOTP_FF()
+        pkt.extended_address = 0
+        pkt.data = b'\x00\x00\x00\x00\x00'
+    else:
+        pkt = ISOTPHeader() / ISOTP_FF()
+        pkt.data = b'\x00\x00\x00\x00\x00\x00'
+
+    pkt.identifier = identifier
+    pkt.message_size = 100
+    return pkt
+
+
+def filter_periodic_packets(packet_dict, verbose=False):
+    """ Filter for periodic packets
+
+    Args:
+        packet_dict: Dictionary with Send-to-ID as key and a tuple
+        (received packet, Recv_ID)
+        verbose: Displays further information
+
+    ISOTP-Filter for periodic packets (same ID, always same timegap)
+    Deletes periodic packets in packet_dict
+    """
+
+    delete_set = set()
+    filter_dict = {}
+
+    for value in packet_dict.values():
+        pkt = value[0]
+        idn = value[1]
+        if idn not in filter_dict.keys():
+            filter_dict[idn] = [pkt]
+        else:
+            filter_dict[idn] += [pkt]
+
+    for idn in filter_dict:
+        pkt_lst = filter_dict[idn]
+        if len(pkt_lst) < 3:
+            continue
+
+        tg = [p1.time - p2.time for p1, p2 in zip(pkt_lst[1:], pkt_lst[:-1])]
+        if all(abs(t1 - t2) < 0.001 for t1, t2 in zip(tg[1:], tg[:-1])):
+            if verbose:
+                print("[i] Identifier 0x%03x seems to be periodic. "
+                      "Filtered.")
+            delete_set.add(idn)
+
+    for key in packet_dict.copy():
+        if packet_dict[key][1] in delete_set:
+            del packet_dict[key]
+
+
+def get_isotp_fc(id_value, id_list, noise_ids, extended, packet,
+                 verbose=False):
+    """Callback for sniff function when packet received
+
+    Args:
+            id_value: packet id of send packet
+            id_list: list of received IDs
+            noise_ids: list of packet IDs which will not be considered when
+                       received during scan
+            extended: boolean if extended scan
+            packet: received packet
+            verbose: displays information during scan
+
+    If received packet is a FlowControl
+    and not in noise_ids
+    append it to id_list
+    """
+    if packet.flags != 0:
```

Comment:
!= 0 un required (except if it can be None ?)

Reply:
shouldn't be None. Thanks